### ISSUE TITLE
[react-native] Components never updates

### ISSUE THREAD

**Issue by Jose4gg** [styled-components/styled-components#1266]
## Version
2.2.2 - native

## Steps to reproduce

1. create a simple `Native` styled component. 

```
const CustomText = styled(Text)`
     color: ${props.light ? "white" : "black};
`
```

2. Change the props in execution with a button or whatever.
3. The component won't update.

## Expected Behavior

The component re-renders

**Comment by hitbear518** [comment]
same problem, 2.2.1 works but 2.2.2 doesn't

**Comment by mxstbr** [comment]
Do you have a full code snippet? This seems like a usage problem, your component should look like this:

```JS
const CustomText = styled(Text)`
     color: ${props => props.light ? "white" : "black};
`
```

> Note the arrow function in the interpolation `props => ...`

And then be used like this:

```JS
<CustomText light={true} />
<CustomText light={false} />
```

**Comment by hitbear518** [comment]
@mxstbr this is my usage that failed after upgrade to 2.2.2 from 2.2.1
```jsx
const NavButtonText = styled.Text`
  color: ${props =>
    props.disabled
      ? 'grey'
      : Platform.select({ ios: '#53a833', android: '#fff' })};
  opacity: ${props => (props.disabled ? 0.7 : 1)};
  font-size: ${navFontSize}px;
  font-family: PingFang-SC-Regular;
`
```

**Comment by Jose4gg** [comment]
@mxstbr  My bad writing the example, I can confirm that my actual code looks exactly like yours but only work on `2.2.1` in `2.2.2` it doesn't

**Comment by mxstbr** [comment]
That's so weird.

/cc @schwers probably a bug in the caching?

**Comment by exentrich** [comment]
Agree, this code not working in `2.2.2`
`${({visible})=>visible?1:0}`

**Comment by rtkhanas** [comment]
Yes, there is probably a bug in 2.2.2. It is possible to reproduce [in this repo](https://github.com/opensource-cards/react-colorizer).

When you drag a handler, left position is never changed since the component is created. The code is [here](https://github.com/opensource-cards/react-colorizer/blob/master/src/components/Handler/components-styled/ColorPickerCircleWrapper/index.native.js).

Everything works well once you change the implementation to:

```
import React from "react";
import { View } from "react-native";
import styled from "styled-components/native";

const ColorPickerCircleWrapper = ({ children, position, size, width }) => (
  <View
    style={{
      alignItems: "center",
      height: size,
      justifyContent: "center",
      left: Math.round(position * width) - size / 2,
      position: "absolute",
      top: 0,
      width: size,
    }}
  >
    {children}
  </View>
);

export default ColorPickerCircleWrapper;
```

**Comment by mxstbr** [comment]
@schwers any ideas?

**Comment by melalj** [comment]
Same here, I took me hours to realize that it was due to the update 2.2.2 (forcing rolling back to 2.2.1 to make it work).
Some components didn't re-rendered while they had an Animated style associated.

**Comment by pencilcheck** [comment]
Same, had a simple circle and the background does not change when the property value changes. Reverting into 2.2.1 fixed it.

**Comment by IljaDaderko** [comment]
I believe I encountered same issue, in my case with animated components, opacity never updates to 1 from 0 in this case: (works with 2.2.1)

```
/* @flow */
import React, { Component } from "react";
import { Animated } from "react-native";
import styled from "styled-components/native";
import { isIphoneX } from "../services/utilities";

// -- types ----------------------------------------------------------------- //
type Props = {
  children: React$Node,
  location: Object
};

type State = {
  previousView: React$Node,
  transition: any
};

// -- styling --------------------------------------------------------------- //
const Container = styled.View`
  padding-top: ${isIphoneX() ? "65px" : "40px"};
  padding-bottom: ${isIphoneX() ? "94px" : "71px"};
  padding-left: 20px;
  padding-right: 20px;
  width: 100%;
  opacity: ${props => props.transition};
`;
const AnimatedContainer = Animated.createAnimatedComponent(Container);

class DashboardRouteAnimation extends Component<Props, State> {
  // -- state --------------------------------------------------------------- //
  state = {
    previousView: null,
    transition: new Animated.Value(0)
  };

  // -- methods ------------------------------------------------------------- //
  componentWillReceiveProps(newProps: Object) {
    if (newProps.location.pathname !== this.props.location.pathname) {
      this.setState({ previousView: this.props.children });
    }
  }

  componentDidMount() {
    Animated.timing(this.state.transition, {
      toValue: 1
    }).start();
  }

  // -- render -------------------------------------------------------------- //
  render() {
    const { children } = this.props;
    const { previousView, transition } = this.state;
    return (
      <AnimatedContainer
        transition={transition.interpolate({
          inputRange: [0, 1],
          outputRange: [0, 1]
        })}
      >
        {previousView || children}
      </AnimatedContainer>
    );
  }
}

export default DashboardRouteAnimation;

```

**Comment by philpl** [comment]
I'll investigate this real quick and try to create a reliable reproduction :smile:

**Comment by philpl** [comment]
A fix is ready in PR #1276; cc @mxstbr

**Comment by IljaDaderko** [comment]
@philpl my use case is still not working, I got workaround going by using `style` prop on `AnimatedContainer`, but in general is my approach supposed to work with styled components?

I tried wrapping `Animated.View` instead of using `createAnimatedComponent`, same result. It updates `transition` prop to correct end value, but without animating it, plus it won't show the component unless other update happens on the view.
